service/controller/v12/resource/instance/create.go

```diff
@@ -112,12 +112,19 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		} else {
 			r.debugger.LogFailedDeployment(ctx, d, err)
 
-			err := r.deleteResourceStatus(customObject, Stage, DeploymentInitialized)
-			if err != nil {
-				return microerror.Mask(err)
+			if key.IsFinalProvisioningState(s) {
```

Comment:
The state could be non final in two cases:
1) deployment is running
2) deployment is failed.
If the deployment is failed, we can retry applying it in the next loop, so we cancel the resource status to make it happen.
If the deployment is running, we shouldn't do anything more until it finishes otherwise we keep updating the deployment at the beginning of the loop and never get to complete the reconciliation.

That's the goal of this if.

Reply:
Is there potential for some infinite loop or deadlock situation?

Reply:
In theory there is, when azure deployment never gets out of the "Running" state.
In that case we will never update that deployment again.
Not sure if this is possible or not (I'd expect at least a timeout on the azure side) and not sure how we could possibly handle this case it anyway.

Reply:
Infinite loops & sort of "deadlocks" here are difficult to handle in code, but we do have already some alerts for them and we can improve them even further when we see what's the right direction.